GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -62,50 +70,67 @@ private void FormDeleteBranchLoad(object sender, EventArgs e)
-                var hasUnmergedBranches = _currentBranch is null || selectedBranches.Any(branch => !_mergedBranches.Contains(branch.Name));
+            MinimumSize = new Size(tlpnlMain.PreferredSize.Width + DpiUtil.Scale(70), height);
+            MaximumSize = new Size(Screen.PrimaryScreen.Bounds.Width, height);
+            Size = new Size(Width, height);
+            ResumeLayout();
+        }
 
-                // we could show yes/no dialog and set forcing checkbox automatically, but more safe way is asking user to do it himself
-                if (hasUnmergedBranches && !ForceDelete.Checked)
+        private void Delete_Click(object sender, EventArgs e)
+        {
+            var selectedBranches = Branches.GetSelectedBranches().ToArray();
+            if (!selectedBranches.Any())
+            {
+                return;
+            }
+
+            if (_currentBranch is not null && selectedBranches.Any(branch => branch.Name == _currentBranch))
+            {
+                MessageBox.Show(this, string.Format(_cannotDeleteCurrentBranchMessage.Text, _currentBranch), _deleteBranchCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return;
+            }
+
+            if (!AppSettings.DontConfirmDeleteBranch)
+            {
+                TaskDialogPage page = new()
                 {
-                    MessageBox.Show(this, _deleteUnmergedBranchForcingSuggestion.Text, _deleteBranchCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
-                }
+                    Text = _deleteBranchQuestion.Text,
+                    Caption = _deleteBranchConfirmTitle.Text,
+                    Icon = TaskDialogIcon.Information,
+                    Buttons = { TaskDialogButton.Yes, TaskDialogButton.No },
+                    Footnote = _useReflogHint.Text,
+                    Verification = new(TranslatedStrings.DontShowAgain),
+                    SizeToContent = true
+                };
 
-                // ask for confirmation to delete unmerged branch that may cause loosing commits
-                // (actually we could check if there are another branches pointing to that commit)
-                if (hasUnmergedBranches
-                    && MessageBox.Show(this, _deleteBranchQuestion.Text, _deleteBranchCaption.Text, MessageBoxButtons.YesNo, MessageBoxIcon.Question) != DialogResult.Yes)
+                bool messageBoxResult = TaskDialog.ShowDialog(Handle, page) == TaskDialogButton.Yes;
+                if (messageBoxResult)
                 {
-                    return;
+                    AppSettings.DontConfirmDeleteBranch = page.Verification.Checked;
                 }
-
-                GitDeleteBranchCmd cmd = new(selectedBranches, ForceDelete.Checked);
-                UICommands.StartCommandLineProcessDialog(this, cmd);
             }
-            catch (Exception ex)
+
+            GitDeleteBranchCmd cmd = new(selectedBranches, force: true);
```

Comment:
```suggestion
            GitDeleteBranchCmd cmd = new(selectedBranches, force: hasUnmergedBranches);
```

Reply:
I want to be able to easily delete local branches irrespective of their merge status. Applying this condition will prevent this.
This is a 'no loss of data' scenario, as it can be restored via reflogs.

Reply:
> I want to be able to easily delete local branches irrespective of their merge status.

Yes, of course. So do I.

> Applying this condition will prevent this.

Why do you think this?
It will add "--force" only if necessary - as @gerhardol requested IIUIC.
(The omitting of "--force" works for checked-out branches but not yet for `detached HEAD` because our parser is buggy: #9279 - to be done after https://github.com/gitextensions/gitextensions/pull/9254#discussion_r649489487. This will not disturb you.)